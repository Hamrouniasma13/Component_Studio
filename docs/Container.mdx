---
name: Container
route: /container
menu: ðŸ§° Technical
---

import { Playground } from 'docz'
import { Container } from '../src'

# Container

`Container` is the main component when it comes to wrap the content on a reasonable size. It's size depend on the media :

- _mobile_ : `100%` of the screen
- _tablet_ : `720px`
- _laptop_ : `960px`
- _desktop_ : `1140px`

It always come with `15px` padding on the left and on the right, and horizontally centered.

<Playground>
  <Container>
    <p>
      Cillum eu laborum dolor occaecat excepteur pariatur. Eu magna veniam
      cupidatat veniam commodo est ad ut irure dolore aliqua. Velit aliqua anim
      fugiat pariatur. Officia eu ex esse sunt anim veniam cillum dolore enim
      pariatur. Pariatur duis nulla adipisicing proident amet tempor in duis
      sunt quis amet quis cupidatat magna. Duis cillum velit fugiat commodo eu.
      Labore anim proident ut ut ipsum.
    </p>
    <p>
      Cillum est aute ex enim est adipisicing enim laboris velit ad eu aliqua
      ea. Mollit consequat voluptate aliquip ea cupidatat non sint consectetur
      cillum. Incididunt irure sit id id officia laboris id aliquip nulla et
      culpa commodo magna labore. Dolor irure dolor labore ipsum culpa. Deserunt
      mollit laborum sunt adipisicing esse exercitation. Non in fugiat officia
      nisi minim nulla aute magna in amet in quis sit aliquip.
    </p>
  </Container>
</Playground>

## Article

`Container` can't be customized, but you can use it's smaller version with the `article` property, which will make it not larger than `798px`.

<Playground>
  <Container article>
    <p>
      Cillum eu laborum dolor occaecat excepteur pariatur. Eu magna veniam
      cupidatat veniam commodo est ad ut irure dolore aliqua. Velit aliqua anim
      fugiat pariatur. Officia eu ex esse sunt anim veniam cillum dolore enim
      pariatur. Pariatur duis nulla adipisicing proident amet tempor in duis
      sunt quis amet quis cupidatat magna. Duis cillum velit fugiat commodo eu.
      Labore anim proident ut ut ipsum.
    </p>
    <p>
      Cillum est aute ex enim est adipisicing enim laboris velit ad eu aliqua
      ea. Mollit consequat voluptate aliquip ea cupidatat non sint consectetur
      cillum. Incididunt irure sit id id officia laboris id aliquip nulla et
      culpa commodo magna labore. Dolor irure dolor labore ipsum culpa. Deserunt
      mollit laborum sunt adipisicing esse exercitation. Non in fugiat officia
      nisi minim nulla aute magna in amet in quis sit aliquip.
    </p>
  </Container>
</Playground>
