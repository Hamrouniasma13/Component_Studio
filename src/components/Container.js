/** @jsx jsx */
import { jsx, ThemeContext } from '@emotion/core'
import React from 'react'

export default function Container({ article, ...props }) {
  const theme = React.useContext(ThemeContext)
  return (
    <div
      css={{
        margin: '0px auto',
        maxWidth: '100%',
        paddingLeft: '15px',
        paddingRight: '15px',
        position: 'relative',

        // in case of float children
        '&:after': {
          display: 'block',
          content: '""',
          clear: 'both'
        },

        [theme.media.tablet]: {
          width: '720px'
        },

        [theme.media.laptop]: {
          width: article ? '798px' : '960px'
        },

        [theme.media.desktop]: {
          width: article ? '798px' : '1140px'
        }
      }}
      {...props}
    />
  )
}
