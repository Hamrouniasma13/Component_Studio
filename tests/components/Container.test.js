import React from 'react';
import renderer from 'react-test-renderer';

import Container from '../../components/Container';

describe('Component: <Container />', () => {
  it('render snapshot', () => {
    const tree = renderer.create(
      <Container>
        Cupidatat tempor minim qui eiusmod cillum nisi officia sint tempor
        dolor.
      </Container>
    );

    expect(tree).toMatchSnapshot();
  });
});
