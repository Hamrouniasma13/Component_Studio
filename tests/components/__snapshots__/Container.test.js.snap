// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Container /> render snapshot 1`] = `
.emotion-0 {
  margin: 0px auto;
  max-width: 100%;
  padding-left: 15px;
  padding-right: 15px;
}

@media only screen and (min-width :768px) {
  .emotion-0 {
    width: 720px;
  }
}

@media only screen and (min-width :992px) {
  .emotion-0 {
    width: 960px;
  }
}

@media only screen and (min-width :1200px) {
  .emotion-0 {
    width: 1140px;
  }
}

<div
  className="emotion-0"
>
  Cupidatat tempor minim qui eiusmod cillum nisi officia sint tempor dolor.
</div>
`;
